scripts/o3de/o3de/manifest.py

```diff
@@ -25,7 +25,10 @@
 # Directory methods
 
 def get_this_engine_path() -> pathlib.Path:
-    return pathlib.Path(os.path.realpath(__file__)).parents[3].resolve()
+    if "SNAP" in os.environ:
+        return pathlib.Path(os.environ.get('SNAP')) / os.environ.get('SNAP_BUILD')
+    else:
+        return pathlib.Path(os.path.realpath(__file__)).parents[3].resolve()
```

Comment:
Why is SNAP specific logic being placed in non-snap code?

The manifest.py should be within the engine root even if it is in a SNAP.

Reply:
This was a workaround for a very specific issue I found with snapd causing this to return an incorrect path, there has been a couple of updates but haven't retested this with the latest version yet.

Reply:
Yes, this was due to the path returned being incorrect when running from the snap container, it was returning a temporary folder and not a path to the o3de snap.

Reply:
Thanks for the clarification.
Could you add that as a comment, that in a snap environment the `__file__` python variable is set to a temporary folder?